src/aleph/vm/controllers/qemu_confidential/instance.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+import logging
+import shutil
+from asyncio.subprocess import Process
+from typing import Callable, Optional
+
+from aleph.vm.conf import settings
+from aleph.vm.controllers.configuration import (
+    Configuration,
+    HypervisorType,
+    save_controller_configuration,
+    QemuConfidentialVMConfiguration,
+)
+from aleph.vm.controllers.qemu import AlephQemuInstance
+from aleph.vm.controllers.qemu.instance import AlephQemuResources, ConfigurationType, logger
+from aleph.vm.network.interfaces import TapInterface
+from aleph_message.models import ItemHash
+from aleph_message.models.execution.environment import MachineResources
+
+logger = logging.getLogger(__name__)
+
+
+class AlephQemuConfidentialResources(AlephQemuResources):
+    pass
+
+
+class AlephQemuConfidentialInstance(AlephQemuInstance):
+    vm_id: int
```

Comment:
Lots of fields are redefined. Does inheritance make sense then or should we rather use a [Protocol](https://docs.python.org/3/library/typing.html#typing.Protocol) ?

Reply:
Doesn't make sense to use a protocol since we want to use the method but I will remove the duplicate field

Reply:
Since it can only be launched via systemd now it will probably make senses at some point to separate what is generating and serializing the config from what is used to launch the emulator. But I don't know enough yet about the firecracker controller   which can be lauched inside and outside systemd to make that split right now.